sherpa/optmethods/optfcts.py

```diff
@@ -20,20 +20,23 @@
 
 """Optimizing functions.
 
-These functions take a callback, the current set of parameters, the
-minimum and maximum parameter ranges, along with optional arguments,
-and return a tuple containing
+As input, these functions take
+  - a callback function, which should return the current statistic value
+    and an array of the statistic value per bin.
+  - the current set of parameters (a numpy array)
+  - the minimum for each parameter (numpy array)
+  - the maximum for each parameter (numpy array)
+  - any optional arguments
 
-    status, parameters, statistic, message, dict
+The return a tuple containing
 
-where ``status`` is a boolean indicating whether the optimisation
-succeeded or not, parameters is the list of parameter values at the
-best-fit location, the statistic value at this location, a string
-message - when ``status`` is `False` this will give information on the
-failure - and a dictionary which depends on the optimiser.
+ - a boolean indicating whether the optimization succeeded or not
+ - the list of parameter values at the best-fit location
+ - the statistic value at this location
+ - a string message - when ``status`` is `False` this will give information
+   on the failure
+ - a dictionary which depends on the optimizer
```

Comment:
I think it's correct to say something like

```suggestion
 - a dictionary which depends on the optimizer and may be empty, or `None`.
```